mlir/lib/AsmParser/Parser.cpp

```diff
@@ -99,6 +99,29 @@ FailureOr<APFloat> detail::parseFloatFromIntegerLiteral(
   return APFloat(semantics, truncatedValue);
 }
 
+FailureOr<APFloat>
+detail::parseFloatFromLiteral(function_ref<InFlightDiagnostic()> emitError,
```

Comment:
Why not add this method to Parser instead of a floating function? All of the users seem to already have a parser instance? That would remove the need for the `emitError` function and make this a bit cleaner.

Reply:
You're right, I didn't notice that all the helper classes have a reference to the parser. That also makes #116171 obsolete.